lib/chef/provider.rb

```diff
@@ -438,13 +438,12 @@ def register_deprecated_lwrp_class(provider_class, class_name)
         if Chef::Provider.const_defined?(class_name, false)
           Chef::Log.warn "Chef::Provider::#{class_name} already exists!  Cannot create deprecation class for #{provider_class}"
         else
-          deprecated_constants[class_name.to_sym] = provider_class
+          Chef::Provider.deprecated_constants[class_name.to_sym] = provider_class
         end
       end
 
-      private
-
       def deprecated_constants
+        raise "Deprecated constants should be called only on Chef::Provider" unless self == Chef::Provider
```

Comment:
This needs an exception type. We should probably enable a RuboCop check for this :)